src/components/all-on-one-form-sub-element-hold-no-modal/all-on-one-form-sub-element-hold-no-modal.ts

```diff
@@ -1,46 +1,85 @@
 import { Component, Input } from '@angular/core';
 import { FaultsScorecardProvider } from '../../providers/faults-scorecard/faults-scorecard';
 import { HazardRecorderProvider } from '../../providers/hazard-recorder/hazard-recorder';
+import { NgRedux } from '@angular-redux/store';
+import { IFaultCounter } from '../button-element/button-element';
+import { Observable } from 'rxjs/Observable';
+import { FaultStoreProvider } from '../../providers/fault-store/fault-store';
 
 @Component({
   selector: 'all-on-one-form-sub-element-hold-no-modal',
   templateUrl: 'all-on-one-form-sub-element-hold-no-modal.html'
 })
 export class AllOnOneFormSubElementHoldNoModalComponent {
 
-  @Input()
-  text: string;
+  @Input('section') section: string = '';
+  @Input() text: string;
+
+  public fault$: Observable<any>;
+
   serious: boolean = false;
   dangerous: boolean = false;
-  counter: number = 0;
 
-  constructor(private faultsService: FaultsScorecardProvider, private hazardRecorderProvider: HazardRecorderProvider) {
+  constructor(private faultsService: FaultsScorecardProvider,
+              private hazardRecorderProvider: HazardRecorderProvider,
+              private ngRedux: NgRedux<IFaultCounter>,
+              private faultStore: FaultStoreProvider) {
+    this.fault$ = this.ngRedux.select(state => 
```

Comment:
We should stick all this redux logic in the service or provider, and subscribe to a returned Observable.